src/ports/postgres/modules/utilities/pivot.py_in

```diff
@@ -58,66 +61,255 @@ def pivot(schema_madlib, source_table, out_table,
                                 pivoted table
         @param aggregate_func   The aggregate function to be applied to the
                                 values
+        @param fill_value       If specified, determines how to fill NULL
+                                values resulting from pivot operation
+        @param keep_null        The flag for determining how to handle NULL
+                                values in pivot columns
     """
-
     """
     Assume we have the following table
         pivset( id INTEGER, piv FLOAT8, val FLOAT8 )
-    where the piv column has 3 distinct values (10.0, 20.0 and 30.0).
+    where the piv column has 3 distinct values (10, 20 and 30).
     If the pivot function call is :
         SELECT madlib.pivot('pivset', 'pivout', 'id', 'piv', 'val');
     We want to construct the following sql code to pivot the table.
         CREATE TABLE pivout AS (SELECT id,
-        sum(CASE WHEN "piv" = '10.0' THEN val ELSE NULL END ) as "piv_10.0",
-        sum(CASE WHEN "piv" = '20.0' THEN val ELSE NULL END ) as "piv_20.0",
-        sum(CASE WHEN "piv" = '30.0' THEN val ELSE NULL END ) as "piv_30.0"
+        avg(CASE WHEN "piv" = '10' THEN val ELSE NULL END ) as "val_avg_piv_10",
+        avg(CASE WHEN "piv" = '20' THEN val ELSE NULL END ) as "val_avg_piv_20",
+        avg(CASE WHEN "piv" = '30' THEN val ELSE NULL END ) as "val_avg_piv_30"
         FROM pivset GROUP BY id ORDER BY id)
 
     """
+
     indices = split_quoted_delimited_str(index)
-    pcol = split_quoted_delimited_str(pivot_cols)
-    pval = split_quoted_delimited_str(pivot_values)
-    validate_pivot_coding(source_table, out_table, indices, pcol, pval)
-    new_col_names =[]
-    sql_list = ["CREATE TABLE " + out_table + " AS (SELECT " + index]
+    pcols = split_quoted_delimited_str(pivot_cols)
+    pvals = split_quoted_delimited_str(pivot_values)
+    validate_pivot_coding(source_table, out_table, indices, pcols, pvals)
 
-    pcol_no_quotes = strip_end_quotes(pcol[0].strip())
-    pval_no_quotes = strip_end_quotes(pval[0].strip())
+    # Strip end quotes from pivot columns
+    pcols = [strip_end_quotes(pcol.strip()) for pcol in pcols]
+    pvals = [strip_end_quotes(pval.strip()) for pval in pvals]
+
+    # Parse the aggregate_func as a dictionary
+    try:
+        param_types = dict.fromkeys(pvals, list)
+        param_types['__madlib_def_pval__'] = list
```

Comment:
It can be removed. I moved the default aggregate but forgot to remove this line.